feos-core/src/state/cache.rs

```diff
@@ -50,6 +50,26 @@ impl Cache {
         }
     }
 
+    pub fn get_or_insert_with_d2_64<F: FnOnce() -> Dual2_64>(
+        &mut self,
+        derivative: Derivative,
+        f: F,
+    ) -> f64 {
+        if let Some(&value) = self.map.get(&PartialDerivative::SecondMixed(derivative, derivative)) {
+            self.hit += 1;
+            value
+        } else {
+            self.miss += 1;
+            let value = f();
+            self.map.insert(PartialDerivative::Zeroth, value.re);
+            self.map
+                .insert(PartialDerivative::First(derivative), value.v1[0]);
+            self.map
+                .insert(PartialDerivative::SecondMixed(derivative, derivative), value.v2[0]);
```

Comment:
I was irritated by the comment in your initial post. Now that it is implemented, it is clear that there is no overlap between the two variants, so using `PartialDerivative::Second` as key for the HashMap is probably simpler to comprehend.

Using both to avoid additional calculations if someone "incorrectly" uses mixed derivatives is probably unnecessary as the relevant functions are not supposed to be used outside of core and the benchmarks anyways.

Reply:
I would not (and do not) use both, but we can use `SecondMixed` exclusively to cover the case of e.g. `PartialDerivative::SecondMixed(DV, DV)`. I am fine with both to be honest :)